### PR TITLE
Re-attach module when the file type is changed

### PR DESCRIPTION
**Pull request by stsewd**
Requires https://github.com/neovim/neovim/pull/14995
Closes https://github.com/nvim-treesitter/nvim-treesitter/issues/1249
Closes https://github.com/nvim-treesitter/nvim-treesitter/issues/1544
Closes https://github.com/nvim-treesitter/nvim-treesitter/issues/1574

### CODE REVIEW

File: lua/nvim-treesitter/configs.lua [hunk 1]
```diff
@@ -442,6 +442,15 @@ function M.detach_module(mod_name, bufnr)
   end
 end
 
+-- Same as attach_module, but if the module is already attached, detach it first.
+-- @param mod_name the module name
+-- @param bufnr the bufnr
+-- @param lang the language of the buffer
+function M.reattach_module(mod_name, bufnr, lang)
+  M.detach_module(mod_name, bufnr)
```
Comment (theHamsta) [inline lua/nvim-treesitter/configs.lua:450]: Can some detach function fail when the module wasn't attached before? Should we add a check whether the module is attached for `detach`?

**Review by theHamsta — APPROVED** [review]

File: lua/nvim-treesitter/configs.lua (hunk 1 above)
Comment (stsewd) [reply lua/nvim-treesitter/configs.lua:450]: The detach function already checks for that https://github.com/nvim-treesitter/nvim-treesitter/blob/3413581b584f8fd124e299b57d77e1ee01886a05/lua/nvim-treesitter/configs.lua#L439-L439

Comment (theHamsta) [reply lua/nvim-treesitter/configs.lua:450]: Someone already planned for that

**Comment by stsewd** [comment]
This should be fine to merge now, but it won't fix all the related problems till the neovim PR is merged.

**Comment by zetashift** [comment]
> This should be fine to merge now, but it won't fix all the related problems till the neovim PR is merged.

This is what I currently see:
image

So for it to apply the TS highlight groups I have defined I need to wait for the neovim PR?

**Comment by stsewd** [comment]
@zetashift how are using that file? If it's being detected as html first, and then it's being changed to something else than yes, you need to wait for the neovim PR.

**Comment by zetashift** [comment]
> @zetashift how are using that file? If it's being detected as html first, and then it's being changed to something else than yes, you need to wait for the neovim PR.

I'm not sure how to check this

`set filetype?` returns `html` and `set syntax?` returns empty.

**Comment by stsewd** [comment]
@zetashift I think that problem isn't related to this PR, feel free to open a new issue describing your problem with an init.vim.

**Comment by zetashift** [comment]
> @zetashift I think that problem isn't related to this PR, feel free to open a new issue describing your problem with a minimal init.vim.

Ah okay, thank you!

**Pull request merged**